modules/reactor/src/meshgenerators/ControlDrumMeshGenerator.C

```diff
@@ -0,0 +1,509 @@
+  auto region_id_int = getElemIntegerFromMesh(*(*_build_mesh), region_id_name);
+  auto assembly_type_id_int = getElemIntegerFromMesh(*(*_build_mesh), assembly_type_id_name);
+
+  unsigned int plane_id_int = 0;
+  if (_extrude)
+    plane_id_int = getElemIntegerFromMesh(*(*_build_mesh), plane_id_name, true);
+
+  // Get next free block ID in mesh in case subdomain ids need to be remapped
+  auto next_block_id = MooseMeshUtils::getNextFreeSubdomainID(*(*(_build_mesh)));
+  std::map<std::string, SubdomainID> rgmb_name_id_map;
+
+  // Loop through all mesh elements and set region ids and reassign block IDs/names
+  // if they were merged during pin stitching
+  for (auto & elem : (*_build_mesh)->active_element_ptr_range())
+  {
+    elem->set_extra_integer(assembly_type_id_int, _assembly_type);
+    const dof_id_type z_id = _extrude ? elem->get_extra_integer(plane_id_int) : 0;
+
+    // Assembly peripheral element (background / duct), set subdomains according
+    // to user preferences and set pin type id to RGMB::MAX_PIN_TYPE_ID - peripheral index
+    // Region id is inferred from z_id and peripheral_idx
+    const auto base_block_id = elem->subdomain_id();
+    const auto base_block_name = (*_build_mesh)->subdomain_name(base_block_id);
+
+    // Check if block name has correct prefix
+    std::string prefix = RGMB::DRUM_BLOCK_NAME_PREFIX + std::to_string(_assembly_type) + "_R";
+    if (!(base_block_name.find(prefix, 0) == 0))
+      continue;
+
+    // Radial index is integer value of substring after prefix
+    const unsigned int radial_idx = std::stoi(base_block_name.substr(prefix.length()));
+
+    // Drum index distinguishes between elements in pad and ex-pad regions that have the same radial
+    // index
+    const unsigned int drum_idx =
+        getDrumIdxFromRadialIdx(radial_idx, elem->true_centroid()(0), elem->true_centroid()(1));
+    subdomain_id_type pin_type = RGMB::MAX_PIN_TYPE_ID - drum_idx;
+    elem->set_extra_integer(pin_type_id_int, pin_type);
+
+    const auto elem_rid = _region_ids[z_id][drum_idx];
+    elem->set_extra_integer(region_id_int, elem_rid);
+
+    // Set element block name and block id
+    auto elem_block_name = default_block_name;
+    if (getReactorParam<bool>(RGMB::region_id_as_block_name))
+      elem_block_name += "_REG" + std::to_string(elem_rid);
+    else if (_has_block_names)
+      elem_block_name += "_" + _block_names[z_id][drum_idx];
+    if (elem->type() == TRI3 || elem->type() == PRISM6)
+      elem_block_name += RGMB::TRI_BLOCK_NAME_SUFFIX;
+    updateElementBlockNameId(
+        *(*_build_mesh), elem, rgmb_name_id_map, elem_block_name, next_block_id);
+  }
+
+  if (getParam<bool>("generate_depletion_id"))
+  {
+    const MooseEnum option = getParam<MooseEnum>("depletion_id_type");
+    addDepletionId(*(*_build_mesh), option, DepletionIDGenerationLevel::Drum, _extrude);
+  }
+
+  (*_build_mesh)->find_neighbors();
```

Comment:
add a comment why you need this. we generally leave the mesh unprepared after most MGs and that s ok, it gets prepared if needed

Reply:
This was done because block ids/names are being updated in generate() for RGMB mesh generators and as a result the neighbors need to be re-determined (see issue and PR here: #21850 and #21851). This was later changed to `set_isnt_prepared` for some RGMB mesh generators (#24018 and #24761). I've gone ahead and changed these calls to `set_isnt_prepared` and added a comment about why this is needed as well